src/SonarScanner.MSBuild.PreProcessor/PreprocessorObjectFactory.cs

```diff
@@ -82,10 +88,15 @@ private async Task<Version> QueryServerVersion(IDownloader downloader)
                 var contents = await downloader.Download(uri);
                 return new Version(contents.Split('-').First());
             }
+            catch (HttpRequestException e) when (e.InnerException is WebException { Status: WebExceptionStatus.ConnectFailure })
+            {
+                logger.LogError(Resources.ERR_UnableToConnectToServer, uri);
```

Comment:
Could it be redundant to redefine the error message rather than using the message provided by the exception?